backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java

```diff
@@ -0,0 +1,236 @@
+        if (exception.getStatus().getCode().value() == Code.PERMISSION_DENIED_VALUE) {
+          LOGGER.warn("CollaboratorResourceMigration Exception: {}", ex.getMessage());
+        } else {
+          LOGGER.warn("CollaboratorResourceMigration Exception: ", ex);
+        }
+      } else {
+        LOGGER.warn("CollaboratorResourceMigration Exception: ", ex);
+      }
+    } finally {
+      ModelDBUtils.unregisteredBackgroundUtilsCount();
+    }
+
+    LOGGER.info("Migration End");
+  }
+
+  private void migrateProjects() {
+    LOGGER.debug("Projects migration started");
+    Long count = getEntityCount(ProjectEntity.class);
+
+    int lowerBound = 0;
+    final int pagesize = this.paginationSize;
+    LOGGER.debug("Total Projects {}", count);
+
+    while (lowerBound < count) {
+
+      try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+        CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+
+        CriteriaQuery<ProjectEntity> criteriaQuery =
+            criteriaBuilder.createQuery(ProjectEntity.class);
+        Root<ProjectEntity> root = criteriaQuery.from(ProjectEntity.class);
+
+        CriteriaQuery<ProjectEntity> selectQuery =
+            criteriaQuery.select(root).orderBy(criteriaBuilder.asc(root.get("id")));
+
+        TypedQuery<ProjectEntity> typedQuery = session.createQuery(selectQuery);
+
+        typedQuery.setFirstResult(lowerBound);
+        typedQuery.setMaxResults(pagesize);
+        List<ProjectEntity> projectEntities = typedQuery.getResultList();
+
+        if (projectEntities.size() > 0) {
+          Set<String> userIds = new HashSet<>();
+          for (ProjectEntity projectEntity : projectEntities) {
+            userIds.add(projectEntity.getOwner());
+          }
+          LOGGER.debug("Project userId list : " + userIds);
+
+          // Fetch the experiment owners userInfo
+          Map<String, UserInfo> userInfoMap =
+              authService.getUserInfoFromAuthServer(userIds, null, null);
+          for (ProjectEntity project : projectEntities) {
+            try {
+              WorkspaceDTO workspaceDTO =
+                  roleService.getWorkspaceDTOByWorkspaceId(
+                      userInfoMap.get(project.getOwner()),
+                      project.getWorkspace(),
+                      project.getWorkspace_type());
+              // if projectVisibility is not equals to ResourceVisibility.ORG_SCOPED_PUBLIC then
+              // ignore the CollaboratorType
+              createWorkspacePermissions(
+                  Optional.of(workspaceDTO.getWorkspaceName()),
+                  Optional.ofNullable(
+                      WorkspaceTypeEnum.WorkspaceType.forNumber(project.getWorkspace_type())),
+                  project.getId(),
+                  Optional.of(Long.parseLong(project.getOwner())),
+                  ModelDBServiceResourceTypes.PROJECT,
+                  CollaboratorTypeEnum.CollaboratorType.READ_WRITE,
```

Comment:
Why are we defaulting to this type of collaborator?

Reply:
So what is expected here as a default? this is adding default when the project has the ORG_DEFAULT workspace.

Reply:
READ_ONLY CollaboratorType is default because READ_ONLY = 0

Reply:
Ok, then i am updating it.

Reply:
Since ORG_CUSTOM is new, we'll never encounter a situation where this field matters in doing the migration, so let's just make sure it doesn't raise an error.